### PR TITLE
Add draw play

### PR DESCRIPTION
**Pull request by davidpn11**
Adds the necessary callbacks and triggers for:
- Users can click the deck if is their turn
- `hasDrawed` validation
- hand listener working perfectly 👍 
- Progression pipe callback for draw plays

### CODE REVIEW

File: functions/src/api/lobby/api/postCreateLobby.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ export const postCreateLobby: RequestHandler<{}, {}, PostBody> = async (
   res
 ) => {
   const { playerId, playerName } = req.body;
+  console.log({ playerId, playerName });
```
Comment (mikelfcosta) [inline functions/src/api/lobby/api/postCreateLobby.ts:21]: log

Comment (mikelfcosta) [inline functions/src/api/lobby/api/postCreateLobby.ts]: Log

File: functions/src/api/cards/api/postBuyCards.ts [hunk 2]
```diff
@@ -51,7 +51,7 @@ export const postBuyCards: RequestHandler<{}, {}, PostBody> = async (
 
   try {
     const updatedPlayer = await buyCards(sessionId, playerId, amount);
-    res.send(updatedPlayer);
+    res.send({ player: updatedPlayer });
```
Comment (mikelfcosta) [inline functions/src/api/cards/api/postBuyCards.ts]: Why this change?

File: web/src/components/HistoryItem/HistoryItem.story.tsx [hunk 3]
```diff
@@ -20,7 +20,7 @@ import {
   Play,
 } from '../../model/Play';
 import { createAvatar, SessionPlayerWithId } from '../../model/Player';
-import { UnionExclude } from '../../model/types';
+// import { UnionExclude } from '../../model/types';
```
Comment (mikelfcosta) [inline web/src/components/HistoryItem/HistoryItem.story.tsx]: comment

File: web/src/views/Game/hooks/useProgressionListener.tsx [hunk 4]
```diff
@@ -54,8 +53,26 @@ export function useProgressionListener() {
         runNextEffect(play);
       }
 
+      function runDrawCardPlayEffect(play: DrawPlay & ID) {
```
Comment (mikelfcosta) [inline web/src/views/Game/hooks/useProgressionListener.tsx:56]: Leaving this for later?

**Review by mikelfcosta — CHANGES_REQUESTED** [review]
Just some adjustments

File: functions/src/api/cards/api/postBuyCards.ts (hunk 2 above)
Comment (davidpn11) [reply functions/src/api/cards/api/postBuyCards.ts]: This change was made considering the scenario where the client would need the player and the card popped.
But this is not required as the existing listeners in the FE are dope 😉  

undoing...

**Review by mikelfcosta — APPROVED** [review]
